dfx/index.ts

```diff
@@ -53,6 +59,61 @@ export async function createAuthenticatedAgent(
     return agent;
 }
 
+/**
+ * Returns an agent authenticated with the identity with the given name.
+ *
+ * IMPORTANT: In order to be synchronous this call will not fetch the root key.
+ * If you are not on mainnet you will need to fetch the root key separately.
+ *
+ * @param identityName
+ * @returns
+ */
+export function getAuthenticatedAgentSync(identityName: string): HttpAgent {
+    const agent = new HttpAgent({
+        host: getAgentHost(),
+        identity: getSecp256k1KeyIdentity(identityName)
+    });
+
+    return agent;
+}
+
+/**
+ * Generates a new identity with the given name if there doesn't already exist
+ * an identity with that name, and returns an agent authenticated with that
+ * identity.
+ *
+ * IMPORTANT: In order to be synchronous this call will not fetch the root key.
+ * If you are not on mainnet you will need to fetch the root key separately.
+ *
+ * @param identityName
+ * @returns
+ */
+export function generateAuthenticatedAgentSync(
```

Comment:
Collapse all down into one function with a default parameter for creating the identity if it does not exist

Reply:
Done